Solutions/Marain.Instance.Deployment/azuredeploy.json

```diff
@@ -55,12 +62,71 @@
           }
         }
       }
+    },
+    {
+      "name": "key-vault",
```

Comment:
So we now have an instance-level key vault? What's this for?

Reply:
It stores the password for a dedicated 'Tenancy Administrator' service principal that is used to authenticate to the tenancy service when initialising it and registering service tenants.

At present a dedicated principal is not strictly necessary as the tenancy service isn't enforcing authorisation checks, though regardless of which identity we use the marain cli tool still requires a suitable secret for its `AzureServicesAuthConnectionString` config setting.